### PR TITLE
Feat - add tags

### PR DESCRIPTION
**Pull request by SinchanaMS**
This feature allows the user to add multiple tags.

**Note: I'm aware I have to move the functions to a different file, the same will be addressed in the next refactoring. Please ignore them for now.**

Files to Check:
- [AddNote.jsx](https://github.com/SinchanaMS/notes-app-react/compare/temp-dev...feat-add-tags#diff-b3a015ff9dc8950fdd347da0338ca86af40a4c5979aa3ad5d4e485dda1217904)
- [Note.jsx](https://github.com/SinchanaMS/notes-app-react/compare/temp-dev...feat-add-tags#diff-3c01810d057a0bb7c7f9c539fbcfac251e5cc99c6b2a15ed543ff80105a4e280)

The other files can be ignored. They were refactored to use a single state to accommodate various lists of notes.

### Preview:

https://user-images.githubusercontent.com/94738532/163675233-c4ace30b-3a0e-4048-8535-776269d2c308.mp4

### CODE REVIEW

File: src/styles/colorpicker.css [hunk 1]
```diff
@@ -3,21 +3,29 @@
...
     position: relative;
   }
   
   .color-wrapper {
+    visibility: hidden;
     width: 10rem;
     display: flex;
     flex-wrap: wrap;
     justify-content: center;
     align-items: center;
```
Comment (srejitk) [inline src/styles/colorpicker.css:19]: You can use a utility class here

File: src/pages/Trash.jsx [hunk 2]
```diff
@@ -5,11 +5,11 @@ import TrashNote from "../components/TrashNote"
...
 export default function Trash() {
 
-    const { trash } = useNote()
+    const { notesData } = useNote()
     
     return (
-        <div>
-            {trash.length !== 0 ? trash.map(item => (
+        <div className='notes-list'>
+            {notesData.trash.length !== 0 ? notesData.trash.map(item => (
```
Comment (srejitk) [inline src/pages/Trash.jsx]: Add optional Chaining here to prevent undefined errors in the future

**Review by srejitk — APPROVED** [review]
Clean code. Really liked your color scheme.

Would suggest limiting the labels to 2-4 Tags to prevent an extensive filter list.

File: src/components/AddNote.jsx [hunk 3]
```diff
@@ -6,47 +6,79 @@ import Editor from './Editor'
...
+            <span class="material-icons md-18 material-icons-outlined add-tag"  onClick={() => addTag(tag)}>add</span>
+            {finalTagsList.map(tag => (
+            <div className='tag-chip'>
+              <p>{tag}</p>
+            </div>
+          ))}
+          </div>
+          <div className='footer-ctas'>
+            <ColorPicker changeColor={(color) => handleColor(note, color)}/>
+            <button className='save-note' type="submit" onClick={()=>addToNotesList(note)}><span class="material-icons md-18 material-icons-outlined">add</span></button>
```
Comment (bhtibrewal) [inline src/components/AddNote.jsx]: Suggestion: you should do this addNote in handle submit

File: src/components/Note.jsx [hunk 4]
```diff
@@ -4,22 +4,23 @@ import Toast from "./Toast"
 
 export default function ({note}) {
 
-  const {setNote, trash, setTrash, notes, setNotes, archives, setArchives} = useNote()
+  const {setNotesData} = useNote()
+  const finalTagsList = [...new Set(note.tags)]
 
-  async function deleteNote(note){
+  const deleteNote = async (note) => {
```
Comment (bhtibrewal) [inline src/components/Note.jsx:10]: Suggestion: you should move this funcyion to a new file

File: src/styles/colorpicker.css [hunk 5]
```diff
@@ -3,21 +3,29 @@
...
     position: absolute;
-    right: 2rem;
+    right: 0rem;
```
Comment (bhtibrewal) [inline src/styles/colorpicker.css]: You dont have to write a unit with zero

**Review by bhtibrewal — APPROVED** [review]
Just written some small suggestions, and you are good to merge

File: src/components/AddNote.jsx (hunk 3 above)
Comment (SinchanaMS) [reply src/components/AddNote.jsx]: Oh yes! I missed changing that, thank you for pointing it out!

**Comment by SinchanaMS** [comment]
Thank you for the reviews, made the necessary changes. Merging to temp-dev.

**Pull request merged**